### PR TITLE
Add the ability to manage tags on ASGs

### PR DESCRIPTION
**Pull request by bobtfish**
Fixes issue #932

I've had a bash at this - code appears to work, but my Go isn't all that great, so please feel free to tear it apart ;)

### CODE REVIEW

File: builtin/providers/aws/resource_aws_autoscaling_group.go [hunk 1]
```diff
@@ -118,10 +119,73 @@ func resourceAwsAutoscalingGroup() *schema.Resource {
...
+						},
+
+						"value": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						"propagate": &schema.Schema{
+							Type:     schema.TypeBool,
+							Required: true,
```
Comment (catsby) [inline builtin/providers/aws/resource_aws_autoscaling_group.go]: The [ASG API reference][1] lists `propagate` as optional, is there a reason to mark this as required here? 


[1]: http://docs.aws.amazon.com/AutoScaling/latest/APIReference/API_CreateAutoScalingGroup.html

**Comment by catsby** [comment]
Looks pretty solid from here.

File: builtin/providers/aws/resource_aws_autoscaling_group.go (hunk 1 above)
Comment (bobtfish) [inline builtin/providers/aws/resource_aws_autoscaling_group.go]: I wasn't sure what default people would like (I assume not propagated), so I made it mandatory so people had to choose.

I'd be happy to make it optional / default to false if that's what people think is best.

**Comment by bobtfish** [comment]
Close in favour of PR #1080

**Pull request closed without merging**